lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1879,14 +1879,143 @@ static Attribute collectFields(MLIRContext *context,
   return ArrayAttr::get(context, operands);
 }
 
+namespace {
```

Comment:
I think we might want to move aggregate forming transforms to their own pass.  (Likewise, incidentally, mux restructuring).  They seem a bit heavy for canonicalization.

Reply:
Sounds good to me.